### PR TITLE
DOC 4059

### PR DESCRIPTION
**Pull request by jamesnocentini**
https://issues.couchbase.com/browse/DOC-4059

### CODE REVIEW

File: modules/ROOT/pages/swift.adoc [hunk 1]
```diff
@@ -236,6 +238,31 @@ The following example enables `verbose` logging for the `replicator` and `query`
...
+
+Console based logging::
+Enabled by default.
+Console based logging is often used to facilitate troubleshooting during development.
+File based and custom logging are disabled by default and can be enabled and configured on the "Database" object.
++
+File based logging::
+File based logging is supported in two formats:
+
+* a plaintext mode
```
Comment (rajagp) [inline modules/ROOT/pages/swift.adoc]: List this after binary mode since thats default

File: modules/ROOT/pages/swift.adoc [hunk 2]
```diff
@@ -236,6 +238,31 @@ The following example enables `verbose` logging for the `replicator` and `query`
...
+* a binary format mode which is much more efficient in terms of storage and performance.
+It is the default configuration for file based logging.
++
+It is recommended that you use binary log format.
+However, in order to view the binary encoded log files, you will need a decoder.
+To decode the binary logs, there is a binary log decoder called the *cbl-log* which is a tool downloadable from https://www.couchbase.com/downloads[here].
```
Comment (rajagp) [inline modules/ROOT/pages/swift.adoc]: You should point this to the docs page which provides the install instructions for cb-log tool . We don't want to take users out of documentation site and no reason for users to have to search through downloads page to just get the link to documentation site.  As FYI - the downloads page  simply has link to the documentation site - nothing more

File: modules/ROOT/pages/swift.adoc [hunk 3]
```diff
@@ -236,6 +238,31 @@ The following example enables `verbose` logging for the `replicator` and `query`
...
++
+Custom logging::
+Provides apps the ability to use other logging frameworks to handle Couchbase Lite log messages.
```
Comment (rajagp) [inline modules/ROOT/pages/swift.adoc]: Please provide an example for each platform. Apps will have to implement this interface
```
interface Logger {
 var level: LogLevel

 func log(
    level: LogLevel,  
    domain: LogDomain,
    message: String)
}
```

Comment (rajagp) [inline modules/ROOT/pages/swift.adoc]:
```suggestion
Provides apps the ability to register callback function in order to receive Couchbase Lite log messages. The messages can be logged using any external logging framework.
```

**Comment by rajagp** [comment]
@jamiltz  done w/ review

File: modules/ROOT/pages/swift.adoc (hunk 1 above)
Comment (jamesnocentini) [reply modules/ROOT/pages/swift.adoc]: https://github.com/couchbase/docs-couchbase-lite/commit/aea18dc924223dd975e2b91ea7b422ad3388f328

File: modules/ROOT/pages/swift.adoc (hunk 2 above)
Comment (jamesnocentini) [reply modules/ROOT/pages/swift.adoc]: https://github.com/couchbase/docs-couchbase-lite/commit/1d30b75458ce604f89054265cb07083c70c2ab3a - FYI the cbl-log section is empty atm (will add content as part of another ticket)

File: modules/ROOT/pages/swift.adoc (hunk 3 above)
Comment (jamesnocentini) [reply modules/ROOT/pages/swift.adoc]: https://github.com/couchbase/docs-couchbase-lite/commit/ac72abaf4f1dbabf202b3e54dd421b353c609ad9 - just a placeholder for now. I have a few questions which we can work through on the docs sync up call.

File: modules/ROOT/examples/csharp/Program.cs [hunk 4]
```diff
@@ -946,4 +962,27 @@ public async Task Send(Message message)
...
         // end::passive-peer-send[]
     }
+
+    // tag::custom-logging[]
+    private class LogTestLogger : ILogger
+    {
+        [NotNull]
+        private readonly List<string> _lines = new List<string>();
+
+        public IReadOnlyList<string> Lines => _lines;
```
Comment (jamesnocentini) [inline modules/ROOT/examples/csharp/Program.cs]: @Sandychuang8 would it be possible to remove references to `lines` to match the Swift version https://github.com/couchbase/docs-couchbase-lite/pull/71/files#diff-dc21fc8be6ee8786054080415959e410R728 ? I believe `lines` is used in unit tests.

File: modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift [hunk 5]
```diff
@@ -57,6 +57,20 @@ class SampleCodeTest {
         Database.setLogLevel(.verbose, domain: .query)
         // end::logging[]
     }
+    
+    func dontTestFileLogging() throws {
+        // tag::file-logging[]
+        let tempFolder = NSTemporaryDirectory().appending("cbllog")
+        Database.log.file.config = LogFileConfiguration.init(directory: tempFolder)
```
Comment (jayahariv) [inline modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift:64]: not much difference, but more easy = `LogFileConfiguration(directory: tempFolder)`

Comment (jamesnocentini) [reply modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift:64]: Yes. Updated in https://github.com/couchbase/docs-couchbase-lite/commit/ae30b84c9253d44e5ed4f11101eec45813c5b6e9

**Pull request merged**